session/session.go

```diff
@@ -310,12 +320,42 @@ func (s *Session) handleTick(ctx context.Context) {
 	s.pm.RecordPeerRequests(nil, live)
 	s.wm.WantBlocks(ctx, live, nil, s.id)
 
-	if len(live) > 0 {
+	// do no find providers on consecutive ticks
+	// -- just rely on periodic rebroadcast
+	if len(live) > 0 && (s.consecutiveTicks == 0) {
 		s.pm.FindMorePeers(ctx, live[0])
 	}
 	s.resetTick()
+
+	if len(s.liveWants) > 0 {
+		s.consecutiveTicks++
+	}
+}
+
+func (s *Session) handleRebroadcast(ctx context.Context) {
```

Comment:
This doesn't actually rebroadcast anything, does it? We should probably just call this something like `handlePeriodicSearch`.